mono/metadata/threads.c

```diff
@@ -4485,24 +4497,42 @@ static void CALLBACK dummy_apc (ULONG_PTR param)
 #endif
 
 /*
- * mono_thread_execute_interruption
+ * mono_thread_execute_interruption_handle
  * 
  * Performs the operation that the requested thread state requires (abort,
  * suspend or stop)
  */
-static MonoException*
-mono_thread_execute_interruption (void)
+static MonoExceptionHandle
+mono_thread_execute_interruption_handle (void)
 {
-	MonoInternalThread *thread = mono_thread_internal_current ();
-	MonoThread *sys_thread = mono_thread_current ();
+	HANDLE_FUNCTION_ENTER ();
 
+	// This function unnecessarily uses the pinned internal thread
+	// via a handle, in order to take advantage of field accessors in handle.h,
+	// against managed pointers within the internal thread.
+	//
+	// Because internal threads are always pinned, a raw pointer
+	// is also used for non-pointer accesses, e.g. integer state, lock, unlock.
+
+	MonoExceptionHandle exc = { 0 };
```

Comment:
`MonoExceptionHandle exc = MONO_HANDLE_NEW (MonoException, NULL)` and subsequently `MONO_HANDLE_ASSIGN(exc, ...)`

Reply:
{ 0 } is a good generic initializer -- pointers, structs, integers, floats, unions sort of.
I didn't want to leave the locals temporarily
uninitialized and was trying to strike a balance
between over-initialization and consistent behavior
and correctness. i.e. I didn't want two news.
I didn't consider new + assign though.
I can do that.
So I guess the "local initialized null" macro is motivated.
I was waffling on that too. This left me w/o needing
another handle.h macro, which I now want again.

Reply:
This still feels like ideal efficiency in some/all cases.

Reply:
Specifically, well, abort_exc was incorrect, and maybe right to just move to a lower scope to fix it to work your way and efficiently.

Reply:
+1 on https://github.com/mono/mono/pull/6639/files#r163303628